custom-docker/build.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+    cat <<EOF 1>&2
+Usage: $0 [-h] [-a <ARCHITECTURES>] [-r <VERSION>]
+Builds the Trino Docker image
+
+-h       Display help
+-a       Build the specified comma-separated architectures, defaults to amd64,arm64,ppc64le
+-t       Image tag name, defaults to trino-custom
+-r       Build the specified Trino release version, downloads all required artifacts
+-x       Skip image tests
+EOF
+}
+
+ARCHITECTURES=(amd64 arm64 ppc64le)
+# also make sure that Dockerfile uses the same version
+TRINO_VERSION='472'
+echo "Trino version set to ${TRINO_VERSION}"
+TAG_PREFIX=trino-custom
+
+SKIP_TESTS=true
+
+function downloadTrinoPlugin() {
+    artifactid=$1
+    version=$2
+    downloadPlugin 'io/trino' ${artifactid} ${version}
+}
+
+function downloadPlugin() {
+    groupPath=$1
+    artifactid=$2
+    version=$3
+    filename="${artifactid}-${version}.zip"
+    url="https://repo1.maven.org/maven2/${groupPath}/${artifactid}/${version}/$filename"
+    echo "Downloading ${url}"
+    curl ${url} -o ${filename}
```

Comment:
Add `--fail` to avoid writing empty files on invalid artifact IDs, and `--location` to handle redirects. Consider `--silent` to disable progress bar but with `--show-error`. I'd suggest using full flag names.